tests/tests_pytorch/core/test_metric_result_integration.py

```diff
@@ -356,7 +356,7 @@ def on_train_epoch_end(self) -> None:
                 assert metrics["callback"]["tracking"] == expected
                 assert computed_value == 2
 
-                assert self.results["training_step.tracking_2"].value == total * devices
+                assert self.results["training_step.tracking_2"].value == total
```

Comment:
The default reduction is mean, and each process is tracking the same batch indices in this logged metric. So if each process has the same value `x`, then the mean will be `x`, not `x * devices`. This didn't surface before, because the test relied on the fact that `result.value` wasn't reduced in-place. But now it is (as with all other reduction functions prior).